src/pybind/cephfs.py

```diff
@@ -317,6 +359,14 @@ def mkdirs(self, path, mode):
         if ret < 0:
             raise make_ex(ret, "error in mkdirs '%s'" % path)
 
+    def rmdir(self, path):
+        self.require_state("mounted")
+        if not isinstance(path, str):
```

Comment:
Yes, cephfs should behave like other linux filesystems and not have any knowledge of the encoding.

The type checking for str was copied from the librados/librbd python bindings where it was added to make sure unicode that could contain '\0' when encoded would not be passed in. In retrospect, I think it was too strict to force callers to convert everything, and auto-converting to utf-8 in the python bindings would be better.